web3/pm.py

```diff
@@ -17,11 +24,98 @@
 # It is not automatically available on a web3 object.
 # To use the `PM` module, attach it to your web3 object
 # i.e. PM.attach(web3, 'pm')
+
+
 class PM(Module):
+
+    def set_registry(self, address=None):
+        if not address:
+            self.registry = Registry.deploy_new_instance(self.web3)
```

Comment:
This is not intuitive behavior imho.  Maybe this should be two different APIs, one that sets it, `set_registry` and one that creates a new one `deploy_and_set_registry`.  Good user guides/examples and documentation are probably the most important part for this since we will have to teach people about the workflows for managing their registry.